docs/features/traffic-management/smi.md

```diff
@@ -0,0 +1,58 @@
+
+- Canary Service: Name of the service that sends traffic only to the canary pods
+- Stable Service: Name of the service that sends traffic only to the stable po ds
+- Root Service: Name of the service that clients use to communicate. If a request comes to this root service not through a proxy, the standard Kubernetes service routing will be used.
+
+Below is an example of a Rollout with all the required fields configured:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: rollout-example
+spec:
+  ...
+  strategy:
+    canary:
+      steps:
+      - setWeight: 5
+      - pause:
+          duration: 600
+      canaryService: canary-svc # required
+      stableService: stable-svc # required
+      trafficRouting:
+        smi:
+         rootService: root-svc # required
+         trafficSplitName: rollout-example-traffic-split # optional
+```
+
+With the above configuration, the controller can automate dynamic traffic splitting. First, the controller manipulates the canary and stable Service listed in the Rollout to make them only receive traffic from the respective canary and stable ReplicaSets. With the stable and canary Services configured, the controller creates a TrafficSplit using these Services in the backend, and the weights of the backend are dynamically configured from the current desired weight of the Rollout's canary steps. With the backends configured, the controller sets the root service of the TrafficSplit to the rootService listed in the Rollout. This configured TrafficSplit along with the Service and Rollout resources enable fine-grained percentages of traffic between two versions of an application. Optionally, the user can specify a name for the traffic split. If there is no name listed in the Rollout, the controller uses the Rollout's name for the TrafficSplit. If a TrafficSplit with that name already exists and isn't owned by that Rollout, the controller marks the Rollout as an error state. 
+
+Here is the TrafficSplit created from the above Rollout:
+
+```yaml
+apiVersion: split.smi-spec.io/v1alpha1
+kind: TrafficSplit
+metadata:
+  name: rollout-example-traffic-split
+spec:
+  service: root-svc
+  backends:
+  - service: stable-svc
+    weight: 5
+  - service: canary-svc
+    weight: 95
+```
+
+As a Rollout progresses through all its steps, the controller updates the TrafficSplit's backend weights to reflect the current weight of the Rollout. When the Rollout has successfully finished executing all the steps, the controller modifies the TrafficSplit to send 100% of traffic to the stable Service.
```

Comment:
You might want to talk through what happens with the labels and how traffic is actually handled under the covers. This is a great overview but I find the picture super helpful for explaining how the system fits together.

Reply:
Great suggestion! I added a picture to make it a little more clear what components are involved. Let me know what you think!

Reply:
Cool! That's super helpful!